core/primitives/src/epoch_manager.rs

```diff
@@ -615,6 +656,28 @@ pub mod epoch_info {
                 let block_producers_sampler = stake_weights(&block_producers_settlement);
                 let chunk_producers_sampler =
                     chunk_producers_settlement.iter().map(|vs| stake_weights(vs)).collect();
+                #[cfg(feature = "protocol_feature_chunk_validation")]
```

Comment:
We should use checked_feature!("protocol_feature_chunk_validation", ChunkValidation, protocol_version) as the condition. This checks not only the feature flag, but that the protocol version is new enough.